### PR TITLE
Feature more complex methods

### PR DESCRIPTION
**Pull request by JoelIvanStella**
## Summary
- Adding more complex methods to Data JAPA
## Trello Card
https://trello.com/c/iYkuq0FH

### CODE REVIEW

**Review by torsello — APPROVED** [review]
GJ! :smiley:

File: src/main/java/com/wolox/training/repositories/BookRepository.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+@Repository
+public interface BookRepository extends JpaRepository<Book, Long> {
+
+    public Optional<Book> findByAuthor(String author);
+
+    public List<Book> findByTitle(String bookTitle);
+
+    public Optional<Book> findByIsbn(String isbn);
+
+    public List<Book> findByPublisherAndGenreAndYear(String publisher, String genre, String year);
```
Comment (jdhiguita) [inline src/main/java/com/wolox/training/repositories/BookRepository.java:19]: What if some parameter is null? I recommend using validation in the query, something like that:
@Query("SELECT b FROM Book b"
        + " WHERE (:publisher is null or b.publisher = :publisher)"
        + " AND (:genre is null or b.genre = :genre)"
        + " AND (:year is null or b.genre = :year)")

**Review by jdhiguita — COMMENTED** [review]
Great job

File: src/main/java/com/wolox/training/repositories/BookRepository.java (hunk 1 above)
Comment (JoelIvanStella) [reply src/main/java/com/wolox/training/repositories/BookRepository.java:19]: Thanks! I develop this feature in the next card, it's ok?

Comment (jdhiguita) [reply src/main/java/com/wolox/training/repositories/BookRepository.java:19]: Yes, sorry I am going fast :)

**Review by jdhiguita — APPROVED** [review]

**Pull request merged**